addons/account/models/chart_template.py

```diff
@@ -196,6 +196,8 @@ def _load(self, template_code, company, install_demo):
             for model in ('account.move',) + TEMPLATE_MODELS[::-1]:
                 if not company.parent_id:
                     self.env[model].sudo().with_context(active_test=False).search([('company_id', 'child_of', company.id)]).with_context({MODULE_UNINSTALL_FLAG: True}).unlink()
+            # Reset company fields that should depend on CoA to their default values.
+            company.write({'anglo_saxon_accounting': False})
```

Comment:
In `_pre_load_data`, add a default value for `anglo_saxon_accounting` instead.
One less write, no weird on/off/on toggles, and it allows for other potential pre-processing if needed

Reply:
Sure. Note that apart from `generic_coa`, all localizations that use anglo-saxon accounting set it in the template values for `res.company`, which are written in the main `_load_data`, so there would still be an on/off/on toggle.